src/content/app/genome-browser/state/browser-general/browserGeneralSlice.ts

```diff
@@ -0,0 +1,301 @@
+import trackPanelStorageService from 'src/content/app/genome-browser/components/track-panel/services/track-panel-storage-service';
+
+import { fetchEnsObject } from 'src/shared/state/ens-object/ensObjectActions';
+
+import {
+  deleteGenomeTrackPanelData,
+  setInitialTrackPanelDataForGenome
+} from 'src/content/app/genome-browser/state/track-panel/trackPanelSlice';
+import {
+  getBrowserActiveEnsObjectIds,
+  getBrowserTrackStates,
+  getBrowserActiveGenomeId,
+  getBrowserActiveEnsObjectId,
+  getChrLocation
+} from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
+import { ensureSpeciesIsEnabled } from 'src/content/app/species-selector/state/speciesSelectorSlice';
+
+import {
+  BrowserTrackStates,
+  TrackActivityStatus
+} from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
+import { RootState } from 'src/store';
+import isEqual from 'lodash/isEqual';
+import { buildFocusIdForUrl } from 'src/shared/state/ens-object/ensObjectHelpers';
+import { getChrLocationStr } from 'src/content/app/genome-browser/helpers/browserHelper';
+
+export type ChrLocation = [string, number, number];
+
+export type ChrLocations = { [genomeId: string]: ChrLocation };
+
+export type BrowserGeneralState = Readonly<{
+  activeGenomeId: string | null;
+  activeEnsObjectIds: { [genomeId: string]: string };
+  trackStates: BrowserTrackStates;
+  chrLocations: ChrLocations; // final location of the browser when user stopped dragging/zooming; used to update the url
+  actualChrLocations: ChrLocations; // transient locations that change while user is dragging or zooming
+  regionEditorActive: boolean;
+  regionFieldActive: boolean;
+  isObjectInDefaultPosition: boolean;
+}>;
+
+export type ParsedUrlPayload = {
+  activeGenomeId: string;
+  activeEnsObjectId: string | null;
+  chrLocation: ChrLocation | null;
+};
+
+export type UpdateTrackStatesPayload = {
+  genomeId: string;
+  categoryName: string;
+  trackId: string;
+  status: TrackActivityStatus;
+};
+
+export const fetchDataForLastVisitedObjects: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = () => async (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const activeEnsObjectIdsMap = getBrowserActiveEnsObjectIds(state);
+  Object.values(activeEnsObjectIdsMap).forEach((objectId) =>
+    dispatch(fetchEnsObject(objectId))
+  );
+};
+
+export const updateTrackStatesAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: BrowserTrackStates) => (dispatch, getState: () => RootState) => {
+  dispatch(updateTrackStates(payload));
+  const trackStates = getBrowserTrackStates(getState());
+  browserStorageService.saveTrackStates(trackStates);
+};
+
+export const setDataFromUrlAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: ParsedUrlPayload) => (dispatch) => {
+  dispatch(setGeneralDataFromUrl(payload));
+  dispatch(setLocationDataFromUrl(payload));
```

Comment:
1) why do we need two dispatches above? Location is now in the same slice.
2) `setGeneralDataFromUrl ` -> just `setDataFromUrl`?

Reply:
Updated it to be just `setDataFromUrl`